File: lang/comparison/task.js
#!/usr/bin/node  --es_staging --harmony --use_strict
let input = new Map()
input.set("A", Array([2.3,"f"], [0.0,"g"], [0.3,"h"]))
input.set("B", Array([0.3001,"i"], [0.01,"j"], [9.3,"kx"]))
input.set("C", Array([1000.7,"l"], [9.3,"m"], [9.2999,"n"]))

let output = new Set()

// forEach can destructure map, for/of can't.
input.forEach( (v, k) => {
	let x = v.filter( (el) => el[0]>=0.3 && el[0]<9.3 );
	for (let e of x) {output.add([k, e[1]])}
});

for (let e of output) {console.log(e)}

File: lang/comparison/task.scala
object Task {
	def main(args: Array[String]) {
		val input = Map(
			"A" -> List((2.3,"f"), (0.01,"g"), (0.3,"h")),
			"B" -> List((0.3001,"i"), (0.01,"j"), (9.3,"kx")),
			"C" -> List((1000.7,"l"), (9.3,"m"), (9.2999,"n")))

		var output = scala.collection.mutable.Set[Tuple2[String,String]]()

		for((k, v) <- input) {
			val x = v.filter( {case (a,b) => a>=0.3 && a<9.3} )
			for(e <- x) output += Tuple2(k, e._2)
		}

		for(e <- output) println(e)
	}
}
